tests/system/conftest.py

```diff
@@ -0,0 +1,61 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import re
+from itertools import chain
+from pathlib import Path
+
+import pytest
+
+OLD_EXECUTOR = os.environ.get("AIRFLOW__CORE__EXECUTOR", default=None)
+REQUIRED_ENV_VARS = ("SYSTEM_TESTS_ENV_ID",)
+
+
+def pytest_configure(config):
+    os.environ["AIRFLOW__CORE__EXECUTOR"] = "DebugExecutor"
+
+
+def pytest_unconfigure(config):
+    if OLD_EXECUTOR is not None:
+        os.environ["AIRFLOW__CORE__EXECUTOR"] = OLD_EXECUTOR
```

Comment:
Should unset "DebugExecutor" here also if the OLD_EXECUTOR is None?

Reply:
I think that with the AIRFLOW__CORE__EXECUTOR set to None the execution will fail anyway. But I could introduce extra flag ie:
```
IS_EXECUTOR_SET = "AIRFLOW__CORE__EXECUTOR" in os.environ
OLD_EXECUTOR = os.environ.get("AIRFLOW__CORE__EXECUTOR", default=None)
(...)
def pytest_unconfigure(config):
    if IS_EXECUTOR_SET:
        os.environ["AIRFLOW__CORE__EXECUTOR"] = OLD_EXECUTOR
```

Reply:
Actually - I think pytest_configure is not the best idea - it will set the executor for the whole pytest execution. I know it would not be likely to have "system" and "non-system" tests - but should not be setting that it in auto-fixture with "package" scope  ?

Reply:
It should - there was reason to use pytest_configure but it was lost in some of the recent designs so it's totally fine (and preffered now) to use auto-fixture. I will do it.

Reply:
And I've checked and actually env variables can be only be strings so OLD_EXECUTOR will not be None as long as the variable existed before (referring to original question by @kosteev)

Reply:
Changed from pytest_configure to auto-fixture

Reply:
@potiuk @kosteev 
Can I resolve this conversation now? Please see the recent changes

Reply:
My concern was about the case when "AIRFLOW__CORE__EXECUTOR" environment variable is not set before (and it means prev_executor will be None), and then "AIRFLOW__CORE__EXECUTOR" will not be unset because of this condition "if prev_executor is not None:".
I do not know the details of execution, maybe this will not happen or not an issue, but technically speaking looks wrong and asks for question. WDYT?

I was thinking about this or something similar.
<pre>
with mock.patch.dict("os.environ", AIRFLOW__CORE__EXECUTOR="DebugExecutor"):
    yield
</pre>

Reply:
Ah you're right, in this case if the env var was not set we should unset it. We will try with mock patching (like in your example).

Reply:
> Ah you're right, in this case if the env var was not set we should unset it. We will try with mock patching (like in your example).

Yep. this is what I also had in mind with yield/patch for autofixture.

Reply:
Updated the code